### ISSUE TITLE
Bundler 2.1.0 breaks bundle install in projects with Gemfile.lock

### ISSUE THREAD

**Issue by stefanhorning** [bundler/bundler#7489]
The new version of bundler released yesterday, seems to break `bundle install` all over the place at our company.
This seems to be the case when there is a pre-existing `Gemfile.lock` in the project that has a `BUNDLED WITH` section. In our case the version in the Gemfile.lock `1.17.3` in most cases. Which leads to the error:

```
/lib/ruby/site_ruby/2.5.0/rubygems.rb:275:in `find_spec_for_exe': Could not find 'bundler' (1.17.3) required by your /vagrant/mpx-db-stats/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:1.17.3
```

The problem can be solved by installing either 1.17.3 with `gem install bundler:1.17.3`, or by removing Gemfile.lock (but not really a good option in some projects).

Also some Travis builds started failing because it uses ruby 2.5.3 by default. Somehow those can be fixed by updating to 2.5.7. However that only helps if Gemfile.lock is already at 2.0.2 I think, as we still had problems with deploys that already build on ruby 2.5.7.

Sorry for this confusing report, but s.th. is weird with this new version and I can't really put my finger on it yet.

**Comment by stefanhorning** [comment]
Looks like I can also reprocude this with other 2.x bundler versions, so perhaps it's not just the latest release. But maybe s.b. else can do better testing here, as we were also in the progress of migrating from ruby 2.5.1 to 2.5.7 currently. The issue only seems to happen on ruby versions < 2.5.7 when the Gemfile.lock is already on 2.x

For the Gemfile.lock being on 1.17.3 cases it's probably expected behaviour and there is no other way arround this then to recreate the `Gemfile.lock` (or stay on old bundler). Still some backwards compatibility would be nice here as it's not easy to get some projects to new `Gemfile.lock`s due to depedencies in there also refereing to bundler etc.

**Comment by deivid-rodriguez** [comment]
Can you run `gem update --system` and try again with a newer version of rubygems?

**Comment by kristenkoyanagi** [comment]
We were able to fix the issue by updating bundler from 1.17.3 to 2.1 and running `bundler update --bundler`. We are using ruby 2.5.3.

**Comment by rezonant** [comment]
This broke our system badly :-( We were bundling with bundler 2.0.2, today any system that pulled bundler 2.1.0 is unable to deploy with the above error :-(

**Comment by deivid-rodriguez** [comment]
Sorry :(

I plan to improve the our bundler switcher for the next version so that this kind of thing don't break anymore in the future and just work.

But for now, the proposed solutions are correct. You should choose a `bundler` version that works for you, and install exactly that version, and then run `bundle update --bundler <version_chosen>` and update your lockfile to use that version.

**Comment by coding-bunny** [comment]
And what if there isn't a `Gemfile.lock`?
Currently we cannot build gems on our CI for testing purposes, because they don't have a lock file, yet depend on old rails 3 for example which fails to build.

* Latest bundler is installed
* bundler 1.17.3 is installed

I expect bundler 2 to fall back automatically when running `bundle install`, yet it doesn't 👍 

```
bundle install
Fetching gem metadata from https://rubygems.org/...........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "bundler":
  In Gemfile:
    bundler (~> 1.0)

  Current Bundler version:
    bundler (2.1.1)
This Gemfile requires a different version of Bundler.
Perhaps you need to update Bundler by running `gem install bundler`?

Could not find gem 'bundler (~> 1.0)' in any of the relevant sources:
  the local ruby installation

```

**Comment by BrianHawley** [comment]
@coding-bunny I ran into the same situation: Had to support rails 3.2 in one mode of a gem project, and I couldn't include the Gemfile.lock because it had to be generated automatically for reasons.

What worked was to install bundler 1.17.3 (we use rvm, so like this):
```
rvm ruby-2.7.0@global do gem install bundler -v 1.17.3
```

Then to use it to create the Gemfile.lock, I ran this code:
```
bundle _1.17.3_ install
```

This generated a Gemfile.lock with this in it:
```
BUNDLED WITH
   1.17.3
```

All subsequent use of that Gemfile.lock worked without needing to specify the bundler version, as long as I had that bundler version installed.

This kind of thing might be worth adding to the docs, @deivid-rodriguez. I found the trick in https://stackoverflow.com/questions/36832689/how-can-i-use-an-older-version-of-bundler-for-a-ruby-project

**Comment by coding-bunny** [comment]
Yeah we are using this workaround as well, but would be smarter for Bundler to just support this for projects, either by specifying something in the `.gemspec` file or so
